wm/frame.go

```diff
@@ -3,271 +3,214 @@
 	} else {
-		f.x += deltaX
-		f.y += deltaY
-	}
-
-	err := xproto.ConfigureWindowChecked(f.wm.x.Conn(), f.id, xproto.ConfigWindowX|xproto.ConfigWindowY|
-		xproto.ConfigWindowWidth|xproto.ConfigWindowHeight,
-		[]uint32{uint32(f.x), uint32(f.y), uint32(f.width), uint32(f.height)}).Check()
-	if err != nil {
-		log.Println("ConfigureWindow Err", err)
+		fr.x += deltaX
+		fr.y += deltaY
 	}
+	fr.updateGeometry(uint16(fr.x), uint16(fr.y), fr.width, fr.height, moveOnly)
 }
 
-func (f *frame) motion(x, y int16) {
+func (fr *frame) motion(x, y int16) {
 	cursor := defaultCursor
-	if x > int16(f.wm.borderWidth()) && x <= int16(f.wm.borderWidth()+f.wm.buttonWidth()) &&
-		y <= int16(f.wm.borderWidth()+f.wm.titleHeight()) {
+	if x > int16(fr.cli.wm.borderWidth()) && x <= int16(fr.cli.wm.borderWidth()+fr.cli.wm.buttonWidth()) &&
+		y <= int16(fr.cli.wm.borderWidth()+fr.cli.wm.titleHeight()) {
 		cursor = closeCursor
-	} else if x >= int16(f.width-f.wm.buttonWidth()) && y >= int16(f.height-f.wm.buttonWidth()) {
+	} else if x >= int16(fr.width-fr.cli.wm.buttonWidth()) && y >= int16(fr.height-fr.cli.wm.buttonWidth()) {
 		cursor = resizeCursor
 	}
 
-	err := xproto.ChangeWindowAttributesChecked(f.wm.x.Conn(), f.id, xproto.CwCursor,
+	err := xproto.ChangeWindowAttributesChecked(fr.cli.wm.x.Conn(), fr.cli.id, xproto.CwCursor,
 		[]uint32{uint32(cursor)}).Check()
 	if err != nil {
-		log.Println("SetCursor Err", err)
+		fyne.LogError("Set Cursor Error", err)
 	}
 }
 
-func (f *frame) RaiseAbove(win desktop.Window) {
-	topID := f.wm.rootID
-	if win != nil {
-		topID = win.(*frame).id
+func (fr *frame) updateGeometry(x uint16, y uint16, w uint16, h uint16, moveOnly bool) {
+	if fr.framed && !moveOnly {
+		borderWidth := 2 * fr.cli.wm.borderWidth()
+		if w < uint16(fr.minWidth)+borderWidth {
+			w = uint16(fr.minWidth) + borderWidth
+		}
+		borderHeight := 2*fr.cli.wm.borderWidth() + fr.cli.wm.titleHeight()
+		if h < uint16(fr.minHeight)+borderHeight {
+			h = uint16(fr.minHeight) + borderHeight
+		}
+		err := xproto.ConfigureWindowChecked(fr.cli.wm.x.Conn(), fr.cli.win, xproto.ConfigWindowX|xproto.ConfigWindowY|
+			xproto.ConfigWindowWidth|xproto.ConfigWindowHeight,
+			[]uint32{uint32(fr.cli.wm.borderWidth()), uint32(fr.cli.wm.borderWidth() + fr.cli.wm.titleHeight()),
+				uint32(w - borderWidth), uint32(h - borderHeight)}).Check()
+		if err != nil {
+			fyne.LogError("Configure Window Error", err)
+		}
+		fr.applyTheme()
 	}
-
-	f.Focus()
-	if f.id == topID {
-		return
+	err := xproto.ConfigureWindowChecked(fr.cli.wm.x.Conn(), fr.cli.id, xproto.ConfigWindowX|xproto.ConfigWindowY|
+		xproto.ConfigWindowWidth|xproto.ConfigWindowHeight,
+		[]uint32{uint32(x), uint32(y), uint32(w), uint32(h)}).Check()
+	if err != nil {
+		fyne.LogError("Configure Window Error", err)
 	}
+}
 
-	f.wm.raiseWinAboveID(f.id, topID)
+func (fr *frame) maximizeFrame() {
+	var w = fr.cli.wm.x.Screen().WidthInPixels
+	var h = fr.cli.wm.x.Screen().HeightInPixels
+	var x, y = 0, 0
```

Comment:
These don't seem to be set - is this a bug?

Reply:
top left of a screen is always 0, 0?

Reply:
Ah yes we need to access the xinerama extension to handle multiple. ignoring for now seems ok

Reply:
As for the 0, 0 - why create the variables at all if it's just 0, 0?